lib/sql-string.js

```diff
@@ -61,6 +61,10 @@ SqlString.escape = function(val, stringifyObjects, timeZone, dialect, field) {
     return val + '';
   }
 
+  if (field && field.type && field.type.validate) {
+    field.type.validate(val);
```

Comment:
Perhaps in query-generator.escape?

Not much better, but at least the qg is our own code, whereas sql-string is adapted from https://www.npmjs.com/package/sql-string so we might want to keep that as free of external modifications as possible